### ISSUE TITLE
fatal: Option -m cannot be combined with -c/-C/-F.

### ISSUE THREAD

**Issue by ExcitedSpider** [gha-utilities/workspace-commit#1]
**Describe the bug**

I got this following the intro in readme:

```
Error: Command failed: git commit -c "user.name gha-utilities" -c "user.email actions@github.com" commit -m "Updates externals" external/figma.common.json
fatal: Option -m cannot be combined with -c/-C/-F.
```

Detail logs can be found [here]( https://github.com/ExcitedSpider/adui-design-token/runs/1700038395?check_suite_focus=true#step:9:20)

**To Reproduce**

```yml
    - name: Add changes to Git tracking
      run: git add -A .
      
    - name: Commit changes
      uses: gha-utilities/workspace-commit@v0.0.3
      with:
        message: Updates externals
        all: true
        
    - name: Push changes
      uses: ad-m/github-push-action@master
      with:
        github_token: ${{ secrets.GITHUB_TOKEN }}
        branch: master
```

**Expected behavior**

Successful commit.

**Screenshots**

![image](https://user-images.githubusercontent.com/35751871/104553064-f3b73f80-5674-11eb-8f45-932a54f5739f.png)

**fix**

Replace with: 
```
- name: config commit username
   run: git config user.name myusername
  
- name: config commit email
   run: git config user.email myemail
      
- name: Commit changes
   run: git commit -m "some message"
```

**Comment by S0AndS0** [comment]
Thanks for the detailed report and providing a fix!

I'll be updating this Action with fixes in the next few days such that `user.name` and `user.email` are set (when defined) prior to commit message step. And I may add scripted checks that inspect if settings are configured with Git when not defined within the Action too.


______


## Notes for future self and readers interested in bug source


Git documentation for `commit` states that `-c`, `-C`, and `-F` options are **not** compatible with `-m`


```bash
git help commit
```


**Help (snip)**


```
# ...
       -m <msg>, --message=<msg>
           Use the given <msg> as the commit message. If multiple -m options are
           given, their values are concatenated as separate paragraphs.

           The -m option is mutually exclusive with -c, -C, and -F.
# ...
```


---


The `git config <setting>` command may be used to check current setting if no override is provided, eg...


```node
const child_process = require('child_process');

const git_user_name = child_process.execSync(
  'git config user.name',
  { encoding: 'utf8' },
);

console.log(`git_user_name -> ${git_user_name}`);
```

**Comment by ExcitedSpider** [comment]
Great, Thx!

**Comment by S0AndS0** [comment]
You're certainly welcome!


---


As of latest tagged release (`v0.0.4`) this bug _should_ be fixed; local tests all show that Action is behaving as intended. And the `ReadMe.md` file is updated with details on how Action behaviour has changed.


**TLDR**


- Setting `user.name` and/or `user.email` prior to this Action will have those configurations take precedence

- Setting `user.name` and/or `user.email` prior to this Action, and overwriting within this action, will have Workspace configurations re-applied after

- Not setting `user.name` or `user.email` will cause hard-coded defaults of `gha-utilities` or `actions@github.com` to be assigned

- Setting `user.name` and/or `user.email` within this Action, without previous assignment, will cause those configurations to persist to following Workspace Actions


---


If for some reason Git still produces `commit` errors about `-c/-C/-F` being incompatible with `-m` options, then please re-open this Issue.
